// This file describes the protobuf for client and backend interaction

syntax = "proto3";

package client;

import "message_definitions.proto";

// Provides general information to client
service CryptoAttacks {
  /**
   * Lists all currently available offensive services
   */
  rpc getAvailableServices(messages.EmptyMessage) returns (messages.AvailableServices) {};
}

// Provides API for attacks on digital signatures
service DigitalSignatureAttack {
  /**
   * Provides function for attack on ECDSA nonce reuse vulnerability
   */
  rpc ecdsaReusedNonceAttack(messages.ReusedNonceAttackRequest)
                                      returns (messages.ReusedNonceAttackResponse) {};
}
